### PR TITLE
Fix mypy for storage-blobs and enable it for CI

### LINKED ISSUE #8366: Type hint/docs for BlobClient.download_blob() is wrong
The type hint for `download_blob` says it returns an iterable of bytes:
https://github.com/Azure/azure-sdk-for-python/blob/04b3a66eb89d8898fb1dd028dd37d1a15a52a398/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L528-L530
In reality, it returns a downloader object –
https://github.com/Azure/azure-sdk-for-python/blob/04b3a66eb89d8898fb1dd028dd37d1a15a52a398/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L599
– which in itself is not iterable-of-bytes either, but you're supposed to call its `.chunks()` method, like so:

```python
                bio = io.BytesIO()
                ssd: StorageStreamDownloader = bc.download_blob()
                for chunk in ssd.chunks():
                    bio.write(chunk)
```

For searchability, the likely error you'll get if you trip on this is

> TypeError: 'StorageStreamDownloader' object is not iterable

Library versions:
```
azure-core               1.0.0     
azure-storage-blob       12.0.0    
```

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes #8366

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: eng/tox/mypy_hard_failure_packages.py [hunk 1]
```diff
@@ -6,5 +6,6 @@
 # --------------------------------------------------------------------------------------------
 
 MYPY_HARD_FAILURE_OPTED = [
-  "azure-core"
+  "azure-core",
+  "azure-storage-blob"
```
Comment (scbedd) [inline eng/tox/mypy_hard_failure_packages.py]: Opt-in looks right to me! Can you confirm that you have seen a MyPy run fail at least once?

Comment (scbedd) [reply eng/tox/mypy_hard_failure_packages.py]: Nevermind, I can see the additional log output confirming it's opt-ed in.

**Comment by rakshith91** [comment]
Working on a different PR

**Pull request closed without merging**